packages/kbn-handlebars/index.ts

```diff
@@ -642,31 +638,44 @@ class ElasticHandlebarsVisitor extends Handlebars.Visitor {
     };
   }
 
-  private setupParams(
-    node: ProcessableNodeWithPathParts,
-    helperName: string
+  private setupDecoratorOptions(
+    decorator: hbs.AST.Decorator | hbs.AST.DecoratorBlock
   ): Handlebars.HelperOptions {
-    const options: Handlebars.HelperOptions = {
-      // @ts-expect-error: Name should be on there, but the offical types doesn't know this
-      name: helperName,
-      hash: this.getHash(node),
-      data: this.runtimeOptions!.data,
-      loc: { start: node.loc.start, end: node.loc.end },
-    };
+    // TypeScript: The types indicate that `decorator.path` technically can be an `hbs.AST.Literal`. However, the upstream codebase always treats it as an `hbs.AST.PathExpression`, so we do too.
+    const name = (decorator.path as hbs.AST.PathExpression).original;
+    const options = this.setupParams(decorator as hbs.AST.DecoratorBlock, name);
 
-    if (node.params.length > 0) {
+    if (decorator.params.length > 0) {
       if (!this.processedRootDecorators) {
         // When processing the root decorators, temporarily remove the root context so it's not accessible to the decorator
         const context = this.scopes.shift();
         // @ts-expect-error: Property 'args' does not exist on type 'HelperOptions'. The 'args' property is expected in decorators
-        options.args = this.resolveNodes(node.params);
+        options.args = this.resolveNodes(decorator.params);
         this.scopes.unshift(context);
       } else {
         // @ts-expect-error: Property 'args' does not exist on type 'HelperOptions'. The 'args' property is expected in decorators
-        options.args = this.resolveNodes(node.params);
+        options.args = this.resolveNodes(decorator.params);
       }
     }
 
+    // @ts-expect-error: Property 'lookupProperty' does not exist on type 'HelperOptions'
+    delete options.lookupProperty; // There's really no tests/documentation on this, but to match the upstream codebase we'll remove `lookupProperty` from the decorator context
```

Comment:
Does this change anything / Would it break if you don't follow original Handlebars?

Reply:
No it doesn't change anything. I think it's just an implementation detail/difference between our implementation and the upstream implementation. But I thought it doesn't hurt to try and match the upstream API. In upstream, `lookupProperty` is added to `options` in a separate step for helpers, whereas we just add it up front when creating the `options` object.